### ISSUE TITLE
monitor running on windows monitoring a soap service does work in one network, but not another

### ISSUE THREAD

**Issue by Roiskia** [louislam/uptime-kuma#5430]
### 📑 I have found these related issues/pull requests

I have not found any simliar issues.

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### Description

I have setup a new uptime kuma instance with one monitor as a bare metal service running on a windows machine but it does not produce any heartbeats with a the correct configuration (afaik).

The soap service i want to monitor is setup with ssl and is reachable through the browser. It is also in use, so i know this side is fine.
I have setup a http monitor for that service with an heartbeat interval of 20 seconds with a known, working request (tested in soapui). Uptime kuma and the soap service are both running on the same machine. 

If i use the fqdn in the url, i wont get any heartbeats from this monitor. If i change the url to use `localhost` instead of the fqdn in the ssl cert, i get the expected errors `Failing; Hostname/IP does not match certificate's altnames (...)` at the expected heartbeat interval.

I have an identical setup on another machine in another, separate network that works fine, so im not sure whats happening here.

Is there a way to log what request are send by uptime kuma and what is happening with them?

### 👟 Reproduction steps

Install runtime and clone repo:
```
winget install OpenJS.NodeJS
winget install Git.Git
winget install CoreyButler.NVMforWindows
git clone https://github.com/louislam/uptime-kuma.git
cd uptime-kuma
$env:NODE_EXTRA_CA_CERTS='C:\path\to\ca\cert.pem'
node server/server.js
```
Configure monitor:
```
monitor type = http(s)
heartbeat interval = 20
retries = 0
heartbeat retry intervall = 20
request timeout = 4
resend notification if down x times = 0
max. redirects = 10
accepted status codes = 200-299
method = post
body encoding = xml
body = <request>
auth method = ntlm
user = user
password = ********
domain = domain
```

### 👀 Expected behavior

Regular heartbeats at the configured interval.

### 😓 Actual Behavior

No heartbeats.

### 🐻 Uptime-Kuma Version

1.23.13

### 💻 Operating System and Arch

Windows Server 2022 x64

### 🌐 Browser

Edge

### 🖥️ Deployment Environment

- Runtime: nodejs 18.20.4 / npm 10.7.0
- Database: sqlite
- Filesystem used to store the database on: Windows
- number of monitors: 1

**Comment by Roiskia** [comment]
Through the logs in the soap service, i found out that it is recieving requests. Somehow uptime kuma isnt showing me these heartbeats though. I had a few instances where the kuma instance seems to freez as the web client lost the connection. After sending the process a ctrl+c in the terminal it resumed to working again. I also got a message about running into the request timeout and the web client updated accordingly this one time. Though i didnt get any further heatbeats until i could reproduce this freez.

**Comment by Roiskia** [comment]
After a few days of observing these systems, i got a few failed heartbeats, as they got restarted. So for some reason it seems like the successfull heartbeats are just ignored. Though i have no idea why this might be the case.

**Comment by CommanderStorm** [comment]
> Install runtime and clone repo:

Please make sure that you follow our [setup guide](https://github.com/louislam/uptime-kuma/wiki/%F0%9F%94%A7-How-to-Install#-non-docker).
`npm run setup` is not present in your version for example


> I have an identical setup on another machine in another, separate network that works fine, so im not sure whats happening here.

Consider bisecting the network to see where your difference is.
Not sure what a soap server exactly is.

**Comment by Roiskia** [comment]
> > Install runtime and clone repo:
> 
> Please make sure that you follow our [setup guide](https://github.com/louislam/uptime-kuma/wiki/%F0%9F%94%A7-How-to-Install#-non-docker). `npm run setup` is not present in your version for example

Yes, you are right, i missed that in my description. I'll check that. Though as far as i understand i would not be able to start the server at all without it since all the dependencies had been missing for example. Or am i wrong?

> > I have an identical setup on another machine in another, separate network that works fine, so im not sure whats happening here.
> Consider bisecting the network to see where your difference is. 

As i wrote on the 11. Dec. 2024 the heartbeats are being send. If they are successfull they are just not recorded in uptime kuma.

> Not sure what a soap server exactly is.

SOAP is an protocoll for doing data exchange and rpc between systems.

**Comment by Roiskia** [comment]
I had time to check the systems and i did use the `npm run setup`. But the hosts that actually run the uptime kuma process does not have internet access. So on another host with internet i did:

```
winget install OpenJS.NodeJS
winget install Git.Git
winget install CoreyButler.NVMforWindows
git clone https://github.com/louislam/uptime-kuma.git
cd uptime-kuma
nvm run setup
```

I then zipped the folder `uptime-kuma`, copied it over to the offline host, installed NodeJS and NVMforWindows manually and executed:

```
$env:NODE_EXTRA_CA_CERTS='C:\path\to\ca\cert.pem'
node server/server.js
```

Does the `nvm run setup` do anything outside the working directory and i therefore broke something by coping just the folder?
